src/foam/dao/index/AATree.js

```diff
@@ -198,7 +186,7 @@ foam.CLASS({
       return s.split(locked).skew(locked);
     },
 
-    function removeKeyValue(key, value, compare, locked) {
+    function removeKeyValue(key, value, compare, locked, nullNode) {
```

Comment:
This was an efficiency optimization you wanted a long time ago (passing things as args rather than shared state). There might be a performance impact from the extra lookup, but as nullNode isn't used all that often compared to the other args I'll have to profile and see. Either way, this API is very tightly coupled to the way TreeIndex works so flexibility gains will be limited.